streams/src/main/java/org/apache/kafka/streams/kstream/internals/CacheEnabledProcessor.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+/**
+ * An interface to mark DSL processor as Cache Enabled
+ */
+public interface CacheEnabledProcessor {
```

Comment:
For all related `KTableXX` operators, we are already creating their associated state store as `ForwardingStateStoreSupplier`, so is this still needed? Or in otherwise, in `TopologyBuilder`, we check both 

```
if (supplier instanceof ForwardingStateStoreSupplier && ((ProcessorNodeFactory) factory).supplier instanceof CacheEnabledProcessor)
```

But is there any scenario where only one of the condition is true?